### ISSUE TITLE
Questions Regarding Reproducing Results and Feature Extraction Settings

### ISSUE THREAD

**Issue by Bariscembakay** [wsi-llava/WSI-LLaVA#12]
Dear Authors,

Thank you for your excellent work. I am currently trying to reproduce your results and would appreciate your clarification on a few points.

First, during inference, I could not get same NLP_metric results with paper (I am getting nearly %5 percent higher than paper in each metric). Which temperature and beam values did you use? Did you set it to 0 for fully deterministic results, or did you use 0.2 as specified in the repository?

Second, I am trying to reproduce the close-ended benchmark so we need those additional .pt files that you have provided. For that reason, I have tried both the Trident and GigaPath recommendation in readme.md and in https://github.com/prov-gigapath/prov-gigapath/blob/main/gigapath/preprocessing/preprocessing.md. While I obtain the expected tensor shapes, the feature values do not match the shared .pt files. 

When using Trident with the GigaPath patch encoder, I used the following arguments for the tile encoder:

--patch_size 256
--batch_size 128
--patch_encoder gigapath
--max_workers 4

Could you please confirm whether these settings are correct, or if there are additional parameters (such as normalization, resizing, color preprocessing) that are required to reproduce the exact features?

Finally, does the Slide Encoder expect the input tile embeddings (from the GigaPath patch encoder) to be L2-normalized, or should they be provided as raw embeddings?

Thank you very much for your time and assistance.

**Comment by hahalonghaha** [comment]
Hi there,
I am using Trident to extract features as well. However, the extracted features have a shape of (768,) instead of (576, xx) and I get .h5 files instead of .pt files. Could you please show how you extract features with Trident or tell me if I made any mistakes with Trident? By the way, I use Titan as the slide encoder in Trident.

**Comment by Bariscembakay** [comment]
> Hi there, I am using Trident to extract features as well. However, the extracted features have a shape of (768,) instead of (576, xx) and I get .h5 files instead of .pt files. Could you please show how you extract features with Trident or tell me if I made any mistakes with Trident? By the way, I use Titan as the slide encoder in Trident.

Hi, 

I am using trident to extract patch features only. I am using gigapath patch encoder with Trident, which outputs a .h5 file. Later, I give these .h5 files into gigapath slide encoder and getting .pt files. I also apply recomended change by authors in the README.md which is: 

Before:
if self.global_pool:
    x = x[:, 1:, :].mean(dim=1)  # global average pooling
    outcome = self.norm(x)

After:
if self.global_pool:
    x = x[:, 1:, :]
    x = torch.nn.functional.adaptive_avg_pool1d(x.permute(0, 2, 1), 576).permute(0, 2, 1)
    outcome = self.norm(x)

You probably use global avarage pooling which is default in Trident. When this change applied, you'll end up with (576, xx) instead of (768,) I guess.
